nmpolicy/internal/capture/resolved.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package capture
+
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
+
+type ResolvedCapture struct {
+	lexer         Lexer
```

Comment:
We store inside the resolver the unmarshaled `desiredStates` `map[string]map[string]interface{}` also since we need them again to do `ResolveCaptureEntryPath` is not that bad.

Note that they are no longer stateless.

Reply:
`resolver.Resolve` can return the unmarshaled version of both the marshaled and unmarshaled.
I think it is more readable over reusing the same resolver. The expander needs just a very specific data/output from the resolver, reusing the same resolver that carries with it the `lexer`, `parser` that are not needed by the expander is an overhead and IMO complicates the code,

Reply:
lexer, parser is not needed by expander is needed for ResolveCaptureEntryPath funciton.